src/analysis/retail/evoker/augmentation/modules/breahtofeons/BreathOfEonsHelper.tsx

```diff
@@ -0,0 +1,538 @@
+import React from 'react';
+import { BreathOfEonsWindows } from './BreathOfEonsRotational';
+import { SubSection } from 'interface/guide';
+import { SpellLink, TooltipElement } from 'interface';
+import { formatDuration, formatNumber } from 'common/format';
+import TALENTS from 'common/TALENTS/evoker';
+import PassFailBar from 'interface/guide/components/PassFailBar';
+import '../Styling.scss';
+import LazyLoadGuideSection from 'analysis/retail/evoker/shared/modules/components/LazyLoadGuideSection';
+import { fetchEvents } from 'common/fetchWclApi';
+import CombatLogParser from '../../CombatLogParser';
+import ExplanationGraph, {
+  DataSeries,
+  GraphData,
+  generateGraphData,
+} from 'analysis/retail/evoker/shared/modules/components/ExplanationGraph';
+import DonutChart from 'parser/ui/DonutChart';
+import { PlayerInfo } from 'parser/core/Player';
+import { DamageEvent } from 'parser/core/Events';
+import { BREATH_OF_EONS_MULTIPLIER, ABILITY_BLACKLIST } from '../../constants';
+
+type Props = {
+  windows: BreathOfEonsWindows[];
+  fightStartTime: number;
+  fightEndTime: number;
+  owner: CombatLogParser;
+};
+
+const debug = false;
+
+const BreathOfEonsHelper: React.FC<Props> = ({ windows, fightStartTime, fightEndTime, owner }) => {
+  const damageTables: {
+    table: DamageEvent[];
+    start: number;
+    end: number;
+  }[] = [];
+
+  /** Generate filter based on black list and whitelist
+   * For now we only look at the players who were buffed
+   * during breath */
+  function getFilter(window: BreathOfEonsWindows) {
+    const playerNames = Array.from(window.breathPerformance.buffedPlayers.keys());
+    const nameFilter = playerNames.map((name) => `"${name}"`).join(', ');
```

Comment:
edge case but in case theres 2 players with the same name, you may want to use the actor.id over actor.name but ... yea

Reply:
swapping to using IDs breaks everything 🙃
https://www.warcraftlogs.com/reports/gfjptA4HWKQ3DrPJ/#fight=15&type=damage-done&pins=2%24Off%24%23244F4B%24expression%24source.id%20in%20(7,%2014,%2017,%2020,%202)%20OR%20source.owner.id%20in%20(7,%2014,%2017,%2020,%202)